packages/client/src/workflow-client.ts

```diff
@@ -92,12 +102,33 @@ import { mapAsyncIterable } from './iterators-utils';
  *   taskQueue: 'tutorial',
  * });
  * await handle.signal(incrementSignal, 2);
- * await handle.query(getValueQuery); // 4
+ * const queryResult = await handle.query(getValueQuery); // 4
+ * const updateResult = await handle.executeUpdate(incrementAndGetValueUpdate, { args: [2] }); // 6
  * await handle.cancel();
- * await handle.result(); // throws WorkflowExecutionCancelledError
+ * await handle.result(); // throws a WorkflowFailedError with `cause` set to a CancelledFailure.
  * ```
  */
 export interface WorkflowHandle<T extends Workflow = Workflow> extends BaseWorkflowHandle<T> {
+  /**
+   * Update a running Workflow.
+   *
+   * @param def an update definition as returned from {@link defineUpdate}
+   *
+   * @example
+   * ```ts
+   * const updateResult = await handle.executeUpdate(incrementAndGetValueUpdate, { args: [2] });
+   * ```
+   */
+  executeUpdate<Ret, Args extends any[] = [], Name extends string = string>(
+    def: UpdateDefinition<Ret, Args, Name> | string,
+    options?: WithArgs<Args, WorkflowUpdateOptions>
+  ): Promise<Ret>;
+
+  startUpdate<Ret, Args extends any[] = [], Name extends string = string>(
+    def: UpdateDefinition<Ret, Args, Name> | string,
+    options?: WithArgs<Args, WorkflowUpdateOptions>
+  ): Promise<WorkflowUpdateHandle<Ret>>;
```

Comment:
As it currently stand, a user may completely omit mandatory `args` if they don't provide the `options` parameter.

Reply:
Yes, `signalWithStart` does it. But that require bringing the type conditional one level higher, which is the opposite direction from `WithArgs<>`.

Reply:
I've changed this so that the options object (containing an `args` key) is required if the update takes args, and otherwise now. The implementation troduces overloads for `executeUpdate` and `startUpdate`.